### PR TITLE
Auth: Respect cache control for JWKS in auth.jwt

### LINKED ISSUE #63465: JWT auth jwk_set_url doesn't respect cache-control header
**What happened**:
When configuring Grafana with JWT Authentication and the JWK Set is taken from url (using `jwk_set_url`), Grafana doesn't respect the `cache_control` http header set by the JWKS endpoint and only follows the configured `cache_ttl` setting.

**What you expected to happen**:
That Grafana honors the `cache-control` header set in the response from the JWKS endpoint and refreshes the JWK set after expiring.

Example response with cache-control header from from [Dex Idp](https://dexidp.io/). Dex by default rotates signing key every 6 hours and immediately starts using the new signing key after rotation, so it is important that client honors the cache-control header in order to resolve the new key.

```
curl -v https://your-auth-provider.example.com/.well-known/jwks.json

< HTTP/2 200
< date: Mon, 20 Feb 2023 15:18:49 GMT
< content-type: application/json
< content-length: 2553
< cache-control: max-age=19250, must-revalidate
< strict-transport-security: max-age=15724800; includeSubDomains
<
{
  "keys": [
    {
      "use": "sig",
      "kty": "RSA",
      "kid": "2f64814115381c2e11632f6b43a042a58665fbf1",
      "alg": "RS256",
      "n": "<omitted>",
      "e": "AQAB"
    },
    ...
]
```

**How to reproduce it (as minimally and precisely as possible)**:

grafana.ini
```
    [auth.jwt]
    enabled = true
    header_name = X-JWT-Assertion
    username_claim = name
    email_claim = email
    jwk_set_url =  https://your-auth-provider.example.com/.well-known/jwks.json
    cache_ttl = 60m
    auto_sign_up = true
    url_login = true
```

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

- Respect cache control for JWKS in auth.jwt if it's lower than configured duration
- Document behavior

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/63465

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/auth/jwt/key_sets.go [hunk 1]
```diff
@@ -141,19 +146,35 @@ func (s *AuthService) initKeySet() error {
...
+		if urlParsed.Scheme != "https" && setting.Env != setting.Dev {
 			return ErrJWTSetURLMustHaveHTTPSScheme
 		}
 		s.keySet = &keySetHTTP{
-			url:             urlStr,
-			log:             s.log,
-			client:          &http.Client{},
+			url: urlStr,
+			log: s.log,
+			client: &http.Client{
```
Comment (IevaVasiljeva) [inline pkg/services/auth/jwt/key_sets.go:161]: question: just curious about why you had to specify these? Which of the default values had to be changed?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM

File: pkg/services/auth/jwt/key_sets.go (hunk 1 above)
Comment (Jguer) [reply pkg/services/auth/jwt/key_sets.go:161]: I did not need to specify, but it's best practice to configure timeouts and the transport (ProxyFromEnv). I got the defaults from some of our other services, actually missing an exported "NewHTTPClient(timeout duration) *http.Client"  somewhere

Comment (IevaVasiljeva) [reply pkg/services/auth/jwt/key_sets.go:161]: So it's mostly to be explicit about the timeouts etc? That makes sense, thanks for the explanation!

**Pull request merged**